solidity/test/VortexBurner.js

```diff
@@ -0,0 +1,691 @@
+const NetworkSettings = contract.fromArtifact('NetworkSettings');
+const VortexBurner = contract.fromArtifact('VortexBurner');
+
+const PPM_RESOLUTION = new BN(1_000_000);
+
+const TKN = new BN(10).pow(new BN(18));
+const RESERVE1_AMOUNT = new BN(100_000_000).mul(TKN);
+const RESERVE2_AMOUNT = new BN(1_000_000).mul(TKN);
+const STANDARD_CONVERTER_WEIGHTS = [500_000, 500_000];
+const TOTAL_SUPPLY = new BN(1_000_000_000).mul(TKN);
+
+describe('VortexBurner', () => {
+    let contractRegistry;
+    let bancorNetwork;
+    let networkToken;
+    let govToken;
+    let govTokenGovernance;
+    let converterRegistry;
+    let converterRegistryData;
+    let networkSettings;
+    let networkFeeWallet;
+    let vortex;
+
+    const owner = defaultSender;
+    const nonOwner = accounts[3];
+
+    before(async () => {
+        contractRegistry = await ContractRegistry.new();
+        bancorNetwork = await BancorNetwork.new(contractRegistry.address);
+
+        const bancorFormula = await BancorFormula.new();
+        await bancorFormula.init();
+
+        await contractRegistry.registerAddress(BANCOR_FORMULA, bancorFormula.address);
+        await contractRegistry.registerAddress(BANCOR_NETWORK, bancorNetwork.address);
+
+        networkSettings = await NetworkSettings.new(defaultSender, new BN(0));
+        await contractRegistry.registerAddress(NETWORK_SETTINGS, networkSettings.address);
+    });
+
+    beforeEach(async () => {
+        converterRegistry = await ConverterRegistry.new(contractRegistry.address);
+        converterRegistryData = await ConverterRegistryData.new(contractRegistry.address);
+        const standardPoolConverterFactory = await StandardPoolConverterFactory.new();
+        const converterFactory = await ConverterFactory.new();
+        await converterFactory.registerTypedConverterFactory(standardPoolConverterFactory.address);
+
+        await contractRegistry.registerAddress(CONVERTER_FACTORY, converterFactory.address);
+        await contractRegistry.registerAddress(CONVERTER_REGISTRY, converterRegistry.address);
+        await contractRegistry.registerAddress(CONVERTER_REGISTRY_DATA, converterRegistryData.address);
+
+        networkFeeWallet = await TokenHolder.new();
+        await networkSettings.setNetworkFeeWallet(networkFeeWallet.address);
+
+        networkToken = await DSToken.new('BNT', 'BNT', 18);
+        await networkToken.issue(owner, TOTAL_SUPPLY);
+
+        govToken = await DSToken.new('vBNT', 'vBNT', 18);
+        await govToken.issue(owner, TOTAL_SUPPLY);
+
+        govTokenGovernance = await TokenGovernance.new(govToken.address);
+        await govToken.transferOwnership(govTokenGovernance.address);
+        await govTokenGovernance.acceptTokenOwnership();
+
+        await networkToken.issue(owner, TOTAL_SUPPLY);
+
+        vortex = await VortexBurner.new(
+            networkToken.address,
+            govTokenGovernance.address,
+            new BN(0),
```

Comment:
We can use native JS numbers when the values that we want to pass are sufficiently small.
I'm not saying that we should use it for every `x <= Number.MAX_SAFE_INTEGER`, but I believe that it is pretty standard when passing `0`.

Reply:
We should reduce the amount of "energy" we spend when writing/reading code, so I'd prefer to keep it `BN` everywhere instead of thinking where should we revert to plain numbers